app/javascript/components/upload/UploadWizard.js

```diff
@@ -79,25 +85,51 @@ export default function UploadWizard({ accession, name }) {
     setFormState(prevFormState => {
       const newFormState = _cloneDeep(prevFormState)
       const fileChanged = newFormState.files.find(f => f._id === fileId)
+      if (!fileChanged) { // we're updating a stale/no-longer existent file -- discard it
+        return prevFormState
+      }
       Object.assign(fileChanged, updates)
       return newFormState
     })
   }
 
   /** save the given file and perform an upload if a selected file is present */
   async function saveFile(file) {
-    updateFile(file._id, { isSaving: true })
-    const fileApiData = formatFileForApi(file)
+    let studyFileId = file._id
+    updateFile(studyFileId, { isSaving: true })
+    const fileSize = file.uploadSelection?.size
+    const isChunked = fileSize > CHUNK_SIZE
```

Comment:
It might not be strictly necessary to chunk upload when the file is small, but unless there's a significant downside to doing so, simply always chunking upload would reduce complexity a fair amount.